### ISSUE TITLE
All text in console and editor loses anti-aliasing

### ISSUE THREAD

**Issue by shivendra90** [spyder-ide/spyder#14780]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [x] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description
I clean installed Miniconda and opened up Spyder. Initially all the fonts were displaying properly, however I wanted to customise some of the fonts from settings. I set `plain text` to Andale Mono and `rich text` to Ubuntu. Rest everything is at defaults.

The moment I hit the 'apply' button, the font seemed to have lost its smoothness. I'm using Ubuntu 20.04 and have restarted, reset and did a clean reinstall of Spyder. But nothing has fixed it. Also opened `jupyter qtconsole` but there it looks normal.

### What steps reproduce the problem?

1. Launch Spyder.
2. Open Spyder's preferences and try set fonts to Andale Mono and Ubuntu.
3. Hit apply.

### What is the expected output? What do you see instead?
The text should not be losing its anti-aliasing. Instead it becomes jittery and unreadable.

### Paste Traceback/Error Below (if applicable)

```python-traceback

NA

```

## Versions

* Spyder version: 4.2.1
* Python version: 3.8.5
* Qt version: 5.9.7
* PyQt version: 5.9.2
* Operating System name/version: Ubuntu 20.04

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.0.2 (OK)
IPython >=7.6.0                 :  7.20.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.0.1 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.7.4 (OK)
pylint >=1.0                    :  2.5.3 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.0             :  0.3.2 (OK)
qdarkstyle >=2.8                :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.1               :  5.0.2 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.7 (OK)
setuptools >=39.0.0             :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                  :  3.5.1 (OK)
spyder_kernels >=1.10.1;<1.11.0 :  1.10.1 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3               :  2.0.1 (OK)
xdg >=0.26                      :  0.26 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  None (NOK)
matplotlib >=2.0.0              :  3.3.4 (OK)
numpy >=1.7                     :  1.19.4 (OK)
pandas >=1.1.1                  :  1.1.4 (OK)
scipy >=0.17.0                  :  1.5.3 (OK)
sympy >=0.7.3                   :  None (NOK)
```

**Comment by andfoy** [comment]
Hi @shivendra90, thanks for reporting this issue, could you please provide a screenshot that displays the problem with the fonts?

**Comment by shivendra90** [comment]
Here's a snapshot with the Andale Mono for plain text and Ubuntu for rich text:

![spyder_report](https://user-images.githubusercontent.com/9760865/108460744-4c46b180-729f-11eb-806c-93ef74e94739.png)

And here's another with Ubuntu Mono for plain text:

![spyder_report2](https://user-images.githubusercontent.com/9760865/108460920-a6e00d80-729f-11eb-88bd-07f69cde5fa9.png)

Ubuntu Mono is more readable but still seems to be lacking some smoothing, especially for normal text other than keywords. It does not seem to be the text that I've used for quite some time in Spyder.

**Comment by andfoy** [comment]
@shivendra90, thanks for the response, could you please provide an example of how should the fonts look like?

**Comment by shivendra90** [comment]
@andfoy Thanks. Here's a snapshot of what I remember resembled to what I used (with Andale Mono):

![spyder_report](https://user-images.githubusercontent.com/9760865/108585802-7caa5080-7370-11eb-8e19-14f7c21622a7.png)

Now this is with Andale Mono bold selected in the inbuilt text editor of Ubuntu, so that is something I figured out. There was no reason to change default fonts in the first place. However because of some errors and utility issues with anaconda, I did a clean install of miniconda instead without removing the caches. While again setting the theme of Spyder to 'Retta' that I am confortable with, the font lost its smoothness. This was not there after the first launch of Spyder in the miniconda base environment.

So what I've tried to do is to set to bold everything in the 'Edit scheme' settings. Strangely, for Andale Mono it stays the same thin font, does not change to bold. Even the keywords that are by default set to bold (and works with Ubuntu Mono) are not appearing bold for Andale Mono. I do not remember how the same font came out as everything bold in the previous installation, although it would be helpful if anyone is able to provide suggestions.

Also is there a way to set plain text to Fira Code? I use that too for better mathematical symbols and ligatures.
